### ISSUE TITLE
[BUG] Cannot mount existing volume

### ISSUE THREAD

**Issue by vfouqueron** [longhorn/longhorn#9133]
## Describe the bug

Volume fails to mount after adding a new node on the cluster, the error message is the following : 'unable to attach volume pvc-38eec084-1cf6-4725-b874-1e62b496d35a to kubernetes-node-f2705c: cannot attach volume pvc-38eec084-1cf6-4725-b874-1e62b496d35a because the data engine image longhornio/longhorn-engine:v1.6.2 is not deployed on at least one of the the replicas' nodes or the node that the volume is going to attach to'

kubernetes-node-f2705c being the new node added. In longhorn-UI all 5 nodes are all schedulable. In the engine-image-ei container on that node, I have only one log : `installed`

## Expected behavior
I can mount the existing node

## Support bundle for troubleshooting

Will be sent by email

## Environment

 - Longhorn version: 1.6.2
 - Impacted volume (PV): pvc-38eec084-1cf6-4725-b874-1e62b496d35a
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): Helm
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version:
   - Number of control plane nodes in the cluster:
   - Number of worker nodes in the cluster:
 - Node config
   - OS type and version:
   - Kernel version:
   - CPU per node: 4
   - Memory per node: 15 GB
   - Disk type (e.g. SSD/NVMe/HDD):
   - Network bandwidth between the nodes (Gbps):
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): Ovh Cloud
 - Number of Longhorn volumes in the cluster: 5

**Comment by vfouqueron** [comment]
Sorry, I had big issues on my cluster, everything went wrong. I had to reinstall completely Longhorn and lost all my volumes in the way...

**Comment by derekbit** [comment]
> Sorry, I had big issues on my cluster, everything went wrong. I had to reinstall completely Longhorn and lost all my volumes in the way...

Really sorry for hearing the news...
How was the issue triggered? If the replica directories still existing. You can back them up and restore the data by manual later.

**Comment by vfouqueron** [comment]
I’ve made errors, I was lock with the issue above, the production environment started to fail and was not able to mount back the volume. It was installed via helm mainly because it deploys with terraform. But in the Rancher interface, the app is listed and it says it has a newer version which is the one on the rancher repo (which is false because it is the same version but bundled by rancher) and even though I have the  longhorn helm repository setup too. I thought that maybe redeploy a slightly different version will maybe reinstall and solve the issue and it did not went well at all. What I understand of the issue is that there were missing CRDs linked to longhorn and everything started to fail, the longhorn manager was logging tons of errors. I had not enough time and comprehension with the issue to start debugging the system, so I reinstalled longhorn completely. I had external backups (not with longhorn because it does not work and I still have no answer to my last message on the issue I opened a month ago), so I already restored manually all backups, I thought it would get the system up way much faster and I know the data would almost certainly be the same than my nightly backup.

I actually don’t know if I will continue with longhorn, the only time I tried to recreate a volume with a longhorn backup (on a dev environment with a setup working in that case), I got a mysterious message about corrupted backup… Nothing is decided yet because I love the product when it works, but when there is an issue, it’s a mess to understand what’s going on.
I got the serious issue (which is partly my fault) and I’m unable to setup something that works despite the massive time I’ve spent trying to understand where was the issue.
